src/pages/Search.tsx

```diff
@@ -147,7 +146,7 @@ const SearchPage: React.FC = () => {
               ).showModal()
             }
           >
-            Click for options
+            {insurance ?? procedure ? insurance && procedure ? "Both filters applied" : "1 filter applied" : "Click for options"}
```

Comment:
can we abstract this more so it's not just dependent on the nested conditionals? future-proofing this code would make it easier to add filters in the future. also, for anyone else reading this i rewrote it in a more verbose fashion, but _it's not a solution._
```suggestion
            let result;
if (insurance !== null && insurance !== undefined) {
 if (procedure !== null && procedure !== undefined) {
   result = "Both filters applied";
 } else {
   result = "1 filter applied";
 }
} else {
 result = "Click for options";
}
```

Reply:
fix

```ts
{(()=>{
   const i = [insurance, procedure].reduce((count, param) => param === undefined ? count : ++count, 0);
   return i === 0 ? "No filters applied" : i !== 1 ? i + " filters applied" : "1 filter applied";
})()}
```